### PR TITLE
Create repo from template

### PR DESCRIPTION
**Pull request by simkimsia**
Based on the comments made in this review https://github.com/PyGithub/PyGithub/pull/2088#pullrequestreview-787467842

i made extra effort to redo the PR so that all the lint related changes are reverted.

This is to solve #1395 and #1974

With the last commit being https://github.com/PyGithub/PyGithub/pull/2090/commits/8b638c5b491b17e974d6e4428d00a2451fdd9c9b

And this is a continuation of the PR in #2012 and #2088

@harismuha123 I have tried to resolve merge conflicts and hope that @s-t-e-v-e-n-k can approve this PR.

### CODE REVIEW

File: tests/Repository.py
Comment (simkimsia) [reply tests/Repository.py:92]: you mean always populated from the API when make a request abt a repo?

File: github/Organization.py
Comment (simkimsia) [reply github/Organization.py]: Resolved in ccb8580

**Comment by simkimsia** [comment]
Sorry @s-t-e-v-e-n-k took me so long to reply as i was busy with other stuff. I have resolved the f-string part. Not too sure abt your second point.

File: tests/Repository.py
Comment (s-t-e-v-e-n-k) [reply tests/Repository.py:92]: I have confirmed it is, so this is fine.

**Comment by s-t-e-v-e-n-k** [comment]
No worries, just need to sort out the new failing tests.

**Comment by s-t-e-v-e-n-k** [comment]
And probably will need to rebase it on master after I merge another PR.

**Comment by simkimsia** [comment]
I am willing to rebase 

Just say the word

**Comment by simkimsia** [comment]
@s-t-e-v-e-n-k  i got this failed test. how do i debug this?

**Comment by s-t-e-v-e-n-k** [comment]
> @s-t-e-v-e-n-k i got this failed test. how do i debug this?

A failure like this means the replay data does not match what the test case is expecting. In this case, what I do is use the -k option for pytest, so I only run that one test, like so: `tox -epy38 -- -k <test name>` and then use pdb to see what's different or tripping up the test.

**Comment by simkimsia** [comment]
> > @s-t-e-v-e-n-k i got this failed test. how do i debug this?
> 
> A failure like this means the replay data does not match what the test case is expecting. In this case, what I do is use the -k option for pytest, so I only run that one test, like so: `tox -epy38 -- -k <test name>` and then use pdb to see what's different or tripping up the test.

do i need to have a separate python venv?

**Comment by s-t-e-v-e-n-k** [comment]
tox creates its own virtualenv

**Comment by simkimsia** [comment]
> ke so: `tox -epy38 -- -k <test name>` and then use pdb to see what's different or tripping up the test.

image

Sorry I think i misunderstood your instructions. Can help me ?

**Comment by s-t-e-v-e-n-k** [comment]
In this case: `tox -epy38 -- -k testCreateRepoFromTemplate` -- this might match more than one test, adding '--co' will help here showing what test are matched. For example:
```
% tox -epy38 -- -k 'testCreateRepo' --co
...
collected 723 items / 716 deselected / 7 selected                              

<Package tests>
  <Module AuthenticatedUser.py>
    <UnitTestCase AuthenticatedUser>
      <TestCaseFunction testCreateRepository>
      <TestCaseFunction testCreateRepositoryWithAllArguments>
      <TestCaseFunction testCreateRepositoryWithAutoInit>
  <Module Organization.py>
    <UnitTestCase Organization>
      <TestCaseFunction testCreateRepoWithAllArguments>
      <TestCaseFunction testCreateRepoWithMinimalArguments>
      <TestCaseFunction testCreateRepositoryWithAutoInit>
  <Module Repository.py>
    <UnitTestCase Repository>
      <TestCaseFunction testCreateRepositoryDispatch>
...
```
Now, this matches more than we want, we only want the first module here, so:

```
%  tox -epy38 -- -k 'AuthenticatedUser and testCreateRepo' --co
...
collected 723 items / 720 deselected / 3 selected                              

<Package tests>
  <Module AuthenticatedUser.py>
    <UnitTestCase AuthenticatedUser>
      <TestCaseFunction testCreateRepository>
      <TestCaseFunction testCreateRepositoryWithAllArguments>
      <TestCaseFunction testCreateRepositoryWithAutoInit>
...
```

Once you're sure what test will be run, and that's the correct test, drop --co and just that one test will run.

**Comment by simkimsia** [comment]
>  tox -epy38 -- -k 'AuthenticatedUser and testCreateRepo'

thank you for advice. you're a great teacher.

i have added the ipdb this way

image


Then i checked the requirements.txt it doesn't include ipdb. so do I add it in myself?

**Comment by s-t-e-v-e-n-k** [comment]
Thank you!

Adding it to requirements.txt will not get tox to install it, best to just install it into the tox venv yourself -- '.tox/py38/bin/pip install ipdb`

**Comment by simkimsia** [comment]
> Adding it to requirements.txt will not get tox to install it, best to just install it into the tox venv yourself -- `.tox/py38/bin/pip install ipdb`

Thank you

This then requires me to run the command as `tox -epy38 -- -k 'AuthenticatedUser and testCreateRepo' -s` else the ipdb won't interrupt as per expected.

I will force push with a new commit

**Comment by simkimsia** [comment]
Sorry @s-t-e-v-e-n-k what's the blocker for merging this? Anything I can do to make this ok?

**Comment by simkimsia** [comment]
Hi @s-t-e-v-e-n-k, what's the blocker for merging this? Anything I can do to make this ok?

**Comment by s-t-e-v-e-n-k** [comment]
@simkimsia I'm sorry, life has gotten in the way of my free time -- lint now fails, `tox -elint` will make the appropriate changes, which you should review and commit. Then if CI is happy, we can merge.

**Comment by simkimsia** [comment]
will look at this later. thank you 🙏

**Comment by simkimsia** [comment]
@s-t-e-v-e-n-k  fixed using `tox -elint` thank you

**Comment by simkimsia** [comment]
@s-t-e-v-e-n-k  any changes i need to do?

**Comment by simkimsia** [comment]
@s-t-e-v-e-n-k  sorry i have to followup. I will follow up in jan 15th next time.

**Comment by simkimsia** [comment]
@s-t-e-v-e-n-k  happy new year. just following up on this. Any changes i need to make?

**Comment by dzierzanowski** [comment]
This is a much needed change, the solution looks solid and simple. Can we get it merged?

Also, I suggest closing the other, now stale PR tackling the same problem: #1395

**Comment by harismuha123** [comment]
@s-t-e-v-e-n-k hello, any news on when this might get merged in?

**Comment by simkimsia** [comment]
I'm formally giving up. If anyone wants to carry on, by all means. No hard feelings. Thank you

**Comment by Ethyling** [comment]
@s-t-e-v-e-n-k Hey, do you think we are ready to merge this? Thank you!

**Comment by StepanKuksenko** [comment]
really want this feature...

**Comment by surgiie** [comment]
@s-t-e-v-e-n-k Do we plan to merge this? would really like this feature.

**Comment by sre-guru** [comment]
Also really need this feature. Can we have this merged please?

**Comment by sre-guru** [comment]
@adamtheturtle @jzelinskie @nhomar @jacquev6 @sfdye @s-t-e-v-e-n-k 
Can someone please approve this. Happy to make any correction to the original PR if author is no longer available/interested.

**Review by sfdye — APPROVED** [review]

**Comment by sfdye** [comment]
Sorry folks for the long wait, I am merging this now. But the new release will probably have to wait until I am back from vacation.

**Comment by simkimsia** [comment]
Thank you @sfdye

**Pull request merged**
